freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -115,6 +120,13 @@ public void deleteServicePrincipal(ServicePrincipalRequest request, String accou
         String canonicalPrincipal = constructCanonicalPrincipal(request.getServiceName(), request.getServerHostName(), realm);
         ipaClient = freeIpaClientFactory.getFreeIpaClientForStack(freeIpaStack);
         ipaClient.deleteService(canonicalPrincipal);
+        //Delete the vault secrets associated with the service principal
+        if (Strings.isNullOrEmpty(request.getClusterCrn())) {
+            LOGGER.warn("Cluster CRN not provided. Auto-generating one");
+            request.setClusterCrn(generateClusterCrn(accountId, request.getEnvironmentCrn()));
```

Comment:
eventually, we want the clusterCRN(some equivalent) to be mandatory. Without it, vault entries associated with a cluster cannot be identified and cleanup when a cluster is deleted.

We did not want to break the API now so we made it optional and are generating one internally.

Reply:
Once this change is merged we will be publishing that cluster crn is expected in the requests. We added a log to see if all the consumers of the API are sending cluster crn in the requests.  If there is anyone who are using the API without the cluster CRN we would know it from the log and the plan is to work with them to adopt it. Once we know that everyone is sending the cluster CRN, it will be made mandatory in the requests.  Bottom line, we have taken this approach to a smooth transition for the consumers of the API.

Reply:
I worked with Kalyan and posted fix for this.